### PR TITLE
Implements Verifiable modes: VOPRF and POPRF

### PR DESCRIPTION
**Pull request by armfazh**
Test vectors from draft-v09 are passing, except for batched vectors.

### CODE REVIEW

**Comment by lukevalenta** [comment]
Should this be rebased on https://github.com/cloudflare/voprf-ts/pull/8 after that's merged?

**Comment by armfazh** [comment]
rebased now

File: src/client.ts [hunk 1]
```diff
@@ -3,34 +3,54 @@
...
     }
 
-    async blind(input: Uint8Array): Promise<{ blind: Blind; blindedElement: Blinded }> {
+    async blind(input: Uint8Array): Promise<[FinalizeData, EvaluationRequest]> {
         const { scalar, blind } = await this.randomBlinder()
-        const dst = Oprf.getHashToGroupDST(this.params.id)
-        const P = await this.params.gg.hashToGroup(input, dst)
-        if (this.params.gg.isIdentity(P)) {
+        const dst = this.getDST(Oprf.LABELS.HashToGroupDST)
+        const P = await this.gg.hashToGroup(input, dst)
```
Comment (lukevalenta) [inline src/client.ts:32]: nit: for consistency with other places in the PR
```suggestion
        const P = await this.gg.hashToGroup(input, this.getDST(Oprf.LABELS.HashToGroupDST))
```

File: src/server.ts [hunk 2]
```diff
@@ -3,68 +3,75 @@
...
-        return new Evaluation(this.params.gg.serialize(Z))
+        return new Evaluation(new Evaluated(this.gg.serialize(Z)))
     }
 
     async fullEvaluate(input: Uint8Array): Promise<Uint8Array> {
-        const dst = Oprf.getHashToGroupDST(this.params.id)
-        const T = await this.params.gg.hashToGroup(input, dst)
-        const issuedElement = new Blinded(this.params.gg.serialize(T))
+        const dst = this.getDST(Oprf.LABELS.HashToGroupDST)
+        const P = await this.gg.hashToGroup(input, dst)
```
Comment (lukevalenta) [inline src/server.ts]: Same as above
```suggestion
        const P = await this.gg.hashToGroup(input, this.getDST(Oprf.LABELS.HashToGroupDST))
```

**Review by lukevalenta — APPROVED** [review]
LGTM!

File: test/vectors.test.ts [hunk 3]
```diff
@@ -24,53 +29,109 @@ function toHex(x: Uint8Array): string {
...
-                // inject the blind value given by the test vector.
-                jest.spyOn(OPRFClient.prototype, 'randomBlinder').mockImplementationOnce(() => {
-                    const blind = new Blind(fromHex(vi.Blind))
-                    const gg = Oprf.getGroup(id)
-                    const scalar = gg.deserializeScalar(blind)
-                    return Promise.resolve({ scalar, blind })
-                })
+                        if (testVector.mode === Oprf.Mode.POPRF) {
+                            const info = fromHex((vi as any).Info as string) // eslint-disable-line @typescript-eslint/no-explicit-any
+                            ;(server as wrapPOPRFServer).info = info
```
[the remark this answers is not in the document]
Comment (armfazh) [reply test/vectors.test.ts:116]: the colon is to separate conflicting statements.
the other part is just a cast.

**Pull request merged**